storage/badger/chat.go

```diff
@@ -0,0 +1,84 @@
+package badger
+
+import (
+	v3Badger "github.com/dgraph-io/badger/v3"
+	"github.com/pastelnetwork/go-commons/errors"
+	"github.com/pastelnetwork/walletnode/storage"
+
+	"github.com/pastelnetwork/go-commons/log"
+)
+
+const (
+	chatLogPrefix = "[chatdb]"
+)
+
+type ChatDB struct {
```

Comment:
I we need to implement a new database based on badger, what will we do? how we are going to separate these databases? 
I think in any case we need at some level to divide these databases by business logic name!

Reply:
Do you mean to avoid conflicts with `keys`?

Reply:
> Do you mean to avoid conflicts with `keys`?

yes, sure!

Reply:
or is it worth storing unrelated data as one database?

Reply:
You have noticed that right, for this purpose, we need to add package `/dao` (DataAccessLevel) with interfaces for every  Objects, e.g.

```go
type Node interface {
	// FindNodes returns all devices
	FindNodes() []Node
	// UpdateNodes updates all data of the nodes in the database
	UpdateNodes(nodes []Node) error

	// FindIfaces returns all ifaces
	FindIfaces() []Iface
	// UpdateIface updates all data of the iface in the database
	UpdateIface(iface Iface) error
}

```
But inside this implementation we will pass `storage interface` memory/badger/whatever, so this way we can change database for entities. but we need that all databases (memory/badger/whatever) use the same interface.

When you integrate your chat you will have to add such similar interface as above.

Do you agree?

Reply:
For example if we need to change db for `Node` all we need it's to pass other database, and don't change `Node` at all.

Reply:
I'm not against it, but I think we can do it easier. since badger is lightweight and big data will not be stored there, I propose to structure it as follows.
Each database that based on badger will be located f.e. in ~/.config/database/<database-name>
They all have to implement one common object and transfer there only the necessary data for the GET, SET, DELETE  operations.

Reply:
> or is it worth storing unrelated data as one database?

Yes, but to avoid conflicts we need to add prefix for keys, e.g. `chat-`

Reply:
or we can leave it as it is, changing the architecture of such a database will not be difficult and for a long time.then, if necessary we can implement an interface on top of f.e. ChatDB structure

Reply:
not   ~/.config/database/ but  ~/.config/database/<database-name>

Reply:
> or we can leave it as it is, changing the architecture of such a database will not be difficult and for a long time.then, if necessary we can implement an interface on top of f.e. ChatDB structure

Right, but I believe `/storage/badger` should be called like now `chatDB` :)

Reply:
great! but what about just pass an instance of dB f.e ChatDB to chat service?

Reply:
You didn't change the name of the struct

Reply:
change to BadgerDB or something else?

Reply:
and we also discussed that each database should have its own structure/directory.

Reply:
Did you decide with Jeff ?

Reply:
decided with you, to avoid conflicts with keys. ok will change to BadgerDB

Reply:
I wrote
>  I don't know what it's better for now, we will decide it later.
But yes, we can decide right now, I think using `structure` is better. because:

Using different db-dir (separate badger db files) for each target. Easier for development, but harder for maintenance, for example if you need to backup files. and if we have for example 20 objects which should be stored in Badger, then we will have 20 db dirs. I don't think this is good solution. Better to have one db badger instance. and store unrelated data there. just take care of using an extra prefix for each key of the object.

Reply:
right, but it will break our global interface of the key-value databases, or we should have a lot of functions to make  prefixes such as 

```
func ChatPrefix(key []byte) []byte {
     return append([]byte("chat_",key...))
}
```
```
func ArtworkPrefix(key []byte) []byte {
     return append([]byte("artwork_",key...))
}
```
I think it is also a bad idea.

Reply:
```go
package dao

import "github.com/pastelnetwork/walletnode/storage"

const dbPrefix = "artwork"

type ArtworkDB struct {
	db storage.KeyValue
}

func (art *ArtworkDB) Set(id string, value []byte) error {
	return art.db.Set(dbPrefix+id, value)
}

func (art *ArtworkDB) Get(id string) ([]byte, error) {
	return art.db.Get(dbPrefix + id)
}
```
this is not the best implementation, i just want to show you that not needed to touch storage interface